src/main/java/com/pixonic/ctop/metrics/AbstractMetrics.java

```diff
@@ -42,33 +48,66 @@ void printMetrics(NavigableSet<ResultItem> readResult, NavigableSet<ResultItem>
         Iterator<ResultItem> readIt = readResult.iterator();
         Iterator<ResultItem> writeIt = writeResult.iterator();
         Long maxReadCount = null, maxWriteCount = null;
-        for(int i = 7; i < height; i++) {
+
+        Long totalReadCount = 0L;
+        Long totalWriteCount = 0L;
+        boolean isMetricsEnabled = !metricsType.equals(MetricsType.NONE);
+
+        for (int i = 7; i < height; i++) {
             if (readIt.hasNext()) {
                 ResultItem resultItem = readIt.next();
                 if (maxReadCount == null) maxReadCount = resultItem.count;
+                totalReadCount += resultItem.count;
                 leftStr = formatCounter(resultItem, maxReadCount);
+
+                //When graphite or console metrics push is enabled then publish it
+                if (isMetricsEnabled) {
+                    String metricString = resultItem.getCf().getKeyProperty("keyspace") + ".table." + resultItem + "." + "read";
+                    metricsCollector.getMetricRegistry().counter(metricString).inc(resultItem.count);
+                }
             } else {
                 leftStr = "";
             }
             if (writeIt.hasNext()) {
                 ResultItem resultItem = writeIt.next();
                 if (maxWriteCount == null) maxWriteCount = resultItem.count;
+                totalWriteCount += resultItem.count;
                 rightStr = formatCounter(resultItem, maxWriteCount);
+
+                //When graphite or console metrics push is enabled then publish it
+                if (isMetricsEnabled) {
+                    String metricString = resultItem.getCf().getKeyProperty("keyspace") + ".table." + resultItem + "." + "write";
+                    metricsCollector.getMetricRegistry().counter(metricString).inc(resultItem.count);
+                }
             } else {
                 rightStr = "";
             }
             if (leftStr.length() == 0 && rightStr.length() == 0) break;
             System.out.println(makeLine(leftStr, rightStr, posWrite));
         }
+
+        //When graphite or console metrics push is enabled then publish the total read and write too
+        if (isMetricsEnabled) {
+            metricsCollector.getReadCount().inc(totalReadCount);
+            metricsCollector.getWriteCount().inc(totalWriteCount);
+        }
     }
 
-    @Override public void shutdown() {
+    @Override
+    public void shutdown() {
         shutdown = true;
     }
 
     protected String formatCounter(ResultItem resultItem, long maxCount) {
         int maxLen = String.valueOf(maxCount).length();
-        return StringUtils.leftPad(String.valueOf(resultItem.count), maxLen + 1) + " " + resultItem;
+
+        //when the mode is not a single Keyspace then prefix the keyspace name to the metric to know from which keyspace it is
+        if (targetType.equals(TargetType.KEYSPACE)) {
```

Comment:
Potential NPE.
To prevent NPE enums can be compared like:
1) ==
2) TargetType.KEYSPACE.equals(targetType)

Reply:
When TargetType(cassandra.target.type) is not given it is by default assumed as ALL(all keyspaces) and it is done in Main.java

`TargetType targetType = TargetType.valueOf(properties.getProperty(Constants.CONFIG_TARGET_TYPE, "ALL"));`